### ISSUE TITLE
Bug in Polyhedron demo menus, with Unity unified menu bar

### ISSUE THREAD

**Issue by gdamiand** [CGAL/cgal#909]
When running the Polyhedron demo, on Ubuntu 15.04 with Unity, after loading an off file, all submenu are empty except the first one (boolean operations). See image below.

![bug](https://cloud.githubusercontent.com/assets/944526/13745618/617a31be-e9ef-11e5-99f8-857937595ec0.png)

**Comment by lrineau** [comment]
It seems that something in the code of the Polyhedron demo triggers an incorrect behavior of the [`libdbusmenu-qt`][1], used by the Unity desktop of Ubuntu.

After a quick look at the code of the Polyhedron demo, I think that there is 95% chances that the bug is in CGAL, and not in `libdbusmenu-qt`. I will discuss with @maxGimeno.

[1]: https://launchpad.net/libdbusmenu-qt

**Comment by maxGimeno** [comment]
@gdamiand  I installed Ubuntu15.04 in a Virtualbox but I still can't reproduce your bug. I tried using appmenu-qt5 but I can't have the same menu than you, what do you use to have qt menu in the window menu, do you know ?

**Comment by maxGimeno** [comment]
@gdamiand I installed a Ubuntu 15.04 in a virtualbox but I don't have the same menus as you, and I think this is why I can't reproduce your bug. Do you know what you use to have the demo menu in the window bar ?

**Comment by gdamiand** [comment]
On ubuntu, run "Paramètres systèmes", click on "apparence", tab "Comportement", here you can choose "afficher les menus pour une fenêtre dans la barre de titre de la fenetre".

**Comment by maxGimeno** [comment]
Ok, I did that but it didn't apply to the demo. I will try to build it under Ubunto and it should work, from what I read on Stackoverflow.

**Comment by maxGimeno** [comment]
It doesn't work. As I cannot reproduce, I suggest you try the patch I wrote in the branch 
https://github.com/CGAL/cgal-dev.git 

and we hope it fixes the bug somehow...

**Comment by gdamiand** [comment]
The bug is still there; and note that I have the same behavior even with menu not in the window but in the global bar.

**Comment by gdamiand** [comment]
I confirm the bug is related with unity; I installed mate and the bug is not there with mate.

**Comment by maxGimeno** [comment]
The global bar ? The one in the top of your screen or the classic Qt menu bar ?

**Comment by lrineau** [comment]
As the bug seems difficult to reproduce, I have decided that it will not block CGAL-4.8.

**Comment by sloriot** [comment]
@gdamiand @maxGimeno  Any progress on this bug? Any chance you can find a solution before the end of the week or shall we postpone to 4.9?

**Comment by maxGimeno** [comment]
I cannot reproduce the issue, so I seriously doubt I can fix it by the end of the week.

**Comment by gdamiand** [comment]
I swith to ubuntu 16.04 (fresh install) and I confirm the bug is still there. @maxGimeno It is not difficult to reproduce the issue: just install an ubuntu...
![bug](https://cloud.githubusercontent.com/assets/944526/15541353/642546b2-228c-11e6-8708-4101a3e87ec3.png)

**Comment by maxGimeno** [comment]
@gdamiand I installed ubuntu in a Virtual Box and I had no bug.

**Comment by gdamiand** [comment]
Try a real install (not with VirtualBox) ?
Moreover this is perhaps related with X-drivers ?

**Comment by lrineau** [comment]
This has a very low priority given that it was reported only once and difficult to reproduce. That is certainly related to something in your installation, Guillaume.

Maxime will not have time to dedicate on that bug, unless you give precise hints on how to reproduce it.

The bug is almost certainly somewhere in the Unity "Menu Proxy" feature. We cannot dedicate time to work around it if at first it is not easy to reproduce.

Meanwhile, the workaround can be to disable the menu proxy for the CGAL Polyhedron demo.
```shell
env UBUNTU_MENUPROXY= ./Polyhedron_3
```
or with `APPMENU_DISPLAY_BOTH=1` instead.

**Comment by gdamiand** [comment]
I understand the low priority.

However I don't think this is related to my installation because as said above I observe the same bug with a fresh install.

**Comment by lrineau** [comment]
Le Wednesday 25 May 2016 07:24:51 Guillaume Damiand a écrit :
> I understand the low priority.
> 
> However I don't think this is related to my installation because as said
> above I observe the same bug with a fresh install.

In the worst case, that may depend on the hardward.

Or maybe that depends:
  - on the set of packages that are chosen during the installation,
  - the desktop environment chosen in the Display Manager at login.

-- 
Laurent Rineau, PhD
R&D Engineer at GeometryFactory           http://www.geometryfactory.com/
Release Manager of the CGAL Project       http://www.cgal.org/

**Comment by sloriot** [comment]
@gdamiand do you still have the issue?

**Comment by gdamiand** [comment]
Yes.

**Comment by lrineau** [comment]
I have good news: the last laptop bought by GeometryFactory (klein.geometryfactory.com, a Dell XPS 13") runs Ubuntu 16.04, and I was impacted by that bug (hopefully, the Operations menu was also available in the context menu).

So, as we can reproduce the issue, there is a chance that we will eventually fix it. My guess is that it will be fixed during the summer. Our schedule in July is rather tight, so probably only in August.

**Comment by gdamiand** [comment]
Good news indeed. Thanks @lrineau.
